wifiphisher/common/victim.py

```diff
@@ -0,0 +1,171 @@
+"""
+This module was made to keep track the information
+of the victims connected to the rogue AP
+"""
+import os
+from macmatcher import MACMatcher as macmatcher
+import wifiphisher.common.constants as constants
+
+
+class Victim(object):
+    """
+    Resembles a Victim (i.e. a connected device to the rogue AP)
+    """
+
+    def __init__(self, vmac_address, ip_address):
+        self.vmac_address = vmac_address
+        self.ip_address = ip_address
+        self.os = "Unknown"
+        self.vendor = "Unknown"
+        victims_instance = Victims.getInstance()
+        victims_instance.add_to_victim_dic(self)
+        self.associate_victim_mac_to_vendor(vmac_address)
+
+    def associate_victim_mac_to_vendor(self, vmac_address):
+        """
+        receives a victims mac address as input, finds the corresponding vendor
+        by using a macmacther object and then accesses the victim
+        dictionary and changes the vendor for the victim with the
+        corresponding mac address
+
+        :param self: empty Victim instance
+        :type self: Victim
+        :param vmac_address: mac address of the victim
+        :type vmac_address: str
+        """
+
+        path_to_mac_file = constants.MAC_PREFIX_FILE
+        macmacther_instance = macmatcher(path_to_mac_file)
+        vendor = macmacther_instance.get_vendor_name(vmac_address)
+        Victims_instance = Victims.getInstance()
+        if vmac_address in Victims_instance.victims_dic.keys():
+            Victims_instance.victims_dic[vmac_address].vendor = vendor
+        else:
+            raise Exception("Error: No such mac address exists in dictionary")
+
+    def assign_ip_to_victim(self, vmac_address, ip_address):
+        """
+        Updates the ip address of the victim with the corresponding mac address
+        """
+
+        Victims_instance = Victims.getInstance()
+        if vmac_address in Victims_instance.victims_dic.keys():
+            Victims_instance.victims_dic[vmac_address].ip_address = ip_address
+        else:
+            raise Exception("Error: No such mac address exists in dictionary")
+
+
+class Victims():
+    """
```

Comment:
I'm not sure we really need this class. We can just define the functions(i.e.` add_to_victim_dic`) and a VICTIM_DICT variable in this module. 

In tui module instead of:
```
victims_instance = victim.Victims.getInstance()
dict_string = victims_instance.get_print_representation()
```
we can just have:
```
dict_string = victim.get_print_representation()
```

Reply:
wont we still need somewhere to have the victim dictionary that holds the Victim objects? we could declare it somewhere and pass it to the Victim.py but i think it looks nice there, if you all agree i could change it